### PR TITLE
Daily RC: Canvas UX - Resize Snap

### PR DESCRIPTION
**Pull request by softmarshmallow**
https://github.com/user-attachments/assets/0e536d60-3101-4a52-b8fb-1c9936e29db4

## Summary by CodeRabbit

* **New Features**
  * Resize snapping: snap to objects and guides with visual hit-point feedback, center-origin mode, and optional aspect-ratio preservation.
  * Scale snap can be force-disabled via Control key (consistent behavior with translate).

* **Tests**
  * Extensive unit and integration tests for continuous resize gestures, multi-anchor snapping, aspect-ratio corner cases, guide snapping, thresholds, and edge cases.

---

> [!NOTE]
> Adds resize snapping to guides and objects with visual feedback, aspect-ratio/center-origin support, and a Control-based force-disable snap for scaling, with full reducer/API integration and tests.
> 
> - **Canvas Core**:
>   - `reducers/tools/snap-resize.ts`: Implement resize snapping (corners-only), guide/object anchors, aspect-ratio and center-origin handling, and visual hit-point mapping.
>   - Integrate snapping into scale gesture in `reducers/methods/transform.ts` via `snapObjectsResize` and threshold logic.
> - **State/Actions/API**:
>   - Add `scale_with_force_disable_snap` to `GestureModifiers`, actions, and editor surface methods (`editor.i.ts`, `action.ts`, `editor.ts`, reducer handling in `reducers/index.ts`).
> - **Hotkeys**:
>   - Update `viewport/hotkeys.tsx` to toggle scaling snap off/on with `Control` (mirrors translate snap toggle) and reset on blur.
> - **Tests**:
>   - New unit, gesture, and integration tests for resize snapping behavior and visuals under various scenarios.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 26d6c6080da35fa14bc9a59d8089ac4d84518489. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: editor/grida-canvas/reducers/tools/snap-resize.ts [hunk 1]
```diff
@@ -0,0 +1,828 @@
...
+          bounding_rect.x + bounding_rect.width,
+          guide.offset,
+        ]);
+        anchor_points.push([
+          bounding_rect.x + bounding_rect.width / 2,
+          guide.offset,
+        ]);
+      }
+    });
+  }
```
[the remark this answers is not in the document]
Comment (softmarshmallow) [reply editor/grida-canvas/reducers/tools/snap-resize.ts:743]: @codesandbot  can you give me a fix for this in plain txt
